docs/pages/config-plugins/mods.mdx

```diff
@@ -311,110 +176,149 @@ export const withIcons = config => {
     [
-      'node_modules/expo-splash-screen/app.plugin.js',
+      'my-config-plugin/index.js',
       <>
-        <YesIcon /> Entry file for custom plugins
+        <YesIcon /> Config Plugin
       </>,
     ],
     [
-      'node_modules/expo-splash-screen/build/index.js',
+      'my-config-plugin/app.plugin.js',
       <>
-        <NoIcon /> Skipped in favor of <code>app.plugin.js</code>
+        <NoIcon /> Skipped outside of a node module
       </>,
     ],
   ]}
 />
 
-```js node_modules/expo-splash-screen/app.plugin.js
-module.exports = config => config;
+#### Raw functions
+
+Expo config objects also support passing functions as-is to the `plugins` array. This is useful for testing, or if you want to use a plugin without creating a file.
+
+```js app.config.js
+const withCustom = (config, props) => config;
+
+const config = {
+  plugins: [
+    [
+      withCustom,
+      {
+        /* props */
+      },
+    ],
+    /* @info Without props */
+    withCustom,
+    /* @end */
+  ],
+};
 ```
 
-### Node module default file
+One caveat to using functions instead of strings is that serialization will replace the function with the function's name. This keeps **manifests** (kind of like the **index.html** for your app) working as expected. Here is what the serialized config would look like:
 
-A config plugin in a node module (without an **app.plugin.js**) will use the `main` file defined in the **package.json**.
+```json
+{
+  "plugins": [["withCustom", {}], "withCustom"]
+}
+```
+
+### Standalone package plugins
+
+Standalone package plugins are published to npm and can be shared across multiple projects, making them ideal for reusable functionality.
+
+> **info** See [Create a module with a config plugin](/config-plugin-and-native-module-tutorial/) for a step-by-step guide on how to create a standalone package plugin.
+
+#### app.plugin.js
+
+Sometimes you want your package to export React components and also support a plugin. To do this, multiple entry points need to be used because the transpilation (Babel preset) could be different. If an **app.plugin.js** file is present in the root of a Node module's directory, it'll be used instead of the package's `main` file.
 
 <FileTree
   files={[
-    ['app.config.js', <code>import "expo-splash-screen"</code>],
-
+    ['app.config.ts', <code>import "expo-splash-screen"</code>],
     ['node_modules/expo-splash-screen', 'Node module'],
     ['node_modules/expo-splash-screen/package.json', <code>"main": "./build/index.js"</code>],
+    [
+      'node_modules/expo-splash-screen/app.plugin.js',
+      <>
+        <YesIcon /> Entry file for custom plugins
+      </>,
+    ],
     [
       'node_modules/expo-splash-screen/build/index.js',
       <>
-        <YesIcon /> Node resolve to this file
+        <NoIcon /> Skipped in favor of <code>app.plugin.js</code>
       </>,
     ],
-
-]}
+  ]}
 />
 
-### Project directory
+#### Node module default file
```

Comment:
riffing off the above note about being prescriptive in the scenarios, I would devote the section-ing to when to use each method, and then describe the resolution order as briefly possible, in a list like "Plugin files in packages are resolved in this order: 1. 2. 3.". Devote most of the space to what folks should do, while also including technical tidbits for those who want to understand the boundaries.